packages/vhd-lib/merge.js

```diff
@@ -115,12 +164,28 @@ module.exports = limitConcurrency(2)(async function merge(
     let counter = 0
 
     const mergeStateWriter = makeThrottledWriter(parentHandler, mergeStatePath, 10e3)
-
     await asyncEach(
       toMerge,
       async blockId => {
         merging.add(blockId)
-        mergeState.mergedDataSize += await parentVhd.coalesceBlock(childVhd, blockId)
+        try {
+          mergeState.mergedDataSize += await parentVhd.coalesceBlock(childVhd, blockId)
+        } catch (error) {
+          if (error.code === 'ENOENT' && isResuming === true) {
+            // when resuming, the blocks moved since the last merge state write are
+            // not in the child anymore but it should be ok
+
+            // it will throw an error if block is missing in parent
+            // won't detect if the block was already in parent and is broken/missing in child
+            const { data } = await parentVhd.readBlock(blockId)
+            assert.strictEqual(data.length, parentVhd.header.blockSize)
```

Comment:
Why not handled in `coalesceBlock`?

Reply:
It wasn't necessary for the file implementation, but maybe it makes sense know.

I feel like handling it above leaks the `Vhd*` implementation details.